system/lpv.py

```diff
@@ -0,0 +1,176 @@
+
+#Gracefully remove physical volume from /dev/sda1
+- lpv: pvs=/dev/sda1 state=absent
+
+#Forcefully remove physical volume even if it's in active use by volume group
+- lpv: pvs=/dev/sda1 state=absent force=yes
+'''
+
+    
+def find_mapper_device_name(module, dm_device):
+        dmsetup_cmd = module.get_bin_path('dmsetup', True)
+        mapper_prefix = '/dev/mapper/'
+        rc, dm_name, err = module.run_command("%s info -C --noheadings -o name %s" % (dmsetup_cmd, dm_device))
+        if rc != 0:
+            module.fail_json(msg="Failed executing dmsetup command.", rc=rc, err=err)
+        mapper_device = mapper_prefix + dm_name.rstrip()
+        return mapper_device
+
+
+def parse_pvs(module, data):
+    pvs = {}
+    dm_prefix = '/dev/dm-'
+    for line in data.splitlines():
+        parts = line.strip().split(';')
+        if parts[0].startswith(dm_prefix):
+            parts[0] = find_mapper_device_name(module, parts[0])
+        pvs[parts[0]] = {
+            'vg_name': parts[1],
+            'pv_size': parts[2],
+            'pv_free': parts[3],
+        }
+    return pvs
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            pvs=dict(type='list', require=True),
+            pv_options=dict(default=''),
+            state=dict(choices=["absent", "present"], default='present'),
+            force=dict(type='bool', default='no'),
+        ),
+        supports_check_mode=True,
+    )
+
+    state = module.params['state']
+    force = module.boolean(module.params['force'])
+    pv_options = module.params['pv_options'].split()
+
+    if module.params['pvs']:
+        dev_list = module.params['pvs']
+    elif state == 'present':
+        module.fail_json(msg="No physical volumes given.")
+
+    # LVM always uses real paths not symlinks so replace symlinks with actual path
+    for idx, dev in enumerate(dev_list):
+        dev_list[idx] = os.path.realpath(dev)
+
+    if state=='present':
+        ### check given devices
+        for test_dev in dev_list:
+            if not os.path.exists(test_dev):
+                module.fail_json(msg="Device %s not found."%test_dev)
+
+    pvs_cmd = module.get_bin_path('pvs', True)
+    pvcreate_cmd = module.get_bin_path('pvcreate', True)
+    pvremove_cmd = module.get_bin_path('pvremove', True)
+
+    rc,current_pvs,err = module.run_command("%s --noheadings -o  pv_name,vg_name,pv_size,pv_free --units b --separator ';'" % pvs_cmd)
+    if rc != 0:
+        module.fail_json(msg="Failed executing pvs command.",rc=rc, err=err)
+    pvs_data = parse_pvs(module, current_pvs)
+
+    changed = False
+
+    if state == 'present':
+        for dev in dev_list:
+            if not dev in pvs_data.keys():
+                if module.check_mode:
+                    module.exit_json(changed=True)
+                rc,_,err = module.run_command([pvcreate_cmd] + pv_options + [dev])
+                if rc == 0:
+                    changed = True
```

Comment:
Exitting here isn't an option, since the module accepts multiple pvs devices. So, it can't exit just after executing `pvcreate` command. there's could be more pvs to be created.